.editorconfig

```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = crlf
+indent_style = space
+indent_size = 4
+insert_final_newline = true
```

Comment:
I like the addition of an `.editorconfig` file. However, here we've said that final newlines should be added to all files. A number of files in this PR (including the `.editorconfig` file itself) don't have a trailing new line. Can we insert these final new lines?